src/lib/bootloader/grub2.rb

```diff
@@ -184,34 +184,18 @@ def disk_order_summary
     def locations
       locations = []
 
-      partition_location = boot_partition_location
-      locations << partition_location unless partition_location.empty?
-      if stage1.extended_partition?
-        # TRANSLATORS: extended is here for extended partition. Keep translation short.
-        locations << Yast::BootStorage.extended_partition.name + _(" (extended)")
-      end
+      partition_location = Yast::BootStorage.boot_partitions.map(&:name).join(", ")
+      locations << partition_location + _(" (/boot)") unless partition_location.empty?
       if stage1.mbr?
         # TRANSLATORS: MBR is acronym for Master Boot Record, if nothing locally specific
         # is used in your language, then keep it as it is.
-        locations << Yast::BootStorage.mbr_disk.name + _(" (MBR)")
+        locations << Yast::BootStorage.boot_disks.map(&:name) + _(" (MBR)")
```

Comment:
beware, you are trying to Array + String here

Reply:
good catch, thanks. I plan to do some intensive testing and hopefully also testsuite will catch it.